### PR TITLE
Update fields rules and examples

### PR DESCRIPTION
**Pull request by bertrand-lorentz**
This adds the co-constraint rules in stage-5.sch, and brings various changes in the other rules.

The notice examples are adapted, and the validation reports are updated accordingly.

This also updates fields.json, to includes various corrections.

### CODE REVIEW

**Review by rouschr — DISMISSED** [review]
Will the examples sdk version be updated to 7 in another pull request?

File: schematrons/stage-5.sch [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+	</rule>
+	<rule context="/*/cac:ProcurementProjectLot[cbc:ID/@schemeName='Lot']/cac:TenderingProcess/cac:EconomicOperatorShortList">
```
Comment (YvesJo) [inline schematrons/stage-5.sch:20]: There is a second set of rules with exactly the same context, hence the second set will never fire

**Review by YvesJo — COMMENTED** [review]
Globally OK. Just wondering for the generation of stage 5 schematron where concurrent sets of rules could be found.

**Comment by bertrand-lorentz** [comment]
> Will the examples sdk version be updated to 7 in another pull request?

Good catch, the notices have been updated for SDK 0.7, so I've added a commit to update their cbc:CustomizationID to "eforms-sdk-0.7".

File: schematrons/stage-5.sch (hunk 1 above)
Comment (bertrand-lorentz) [reply schematrons/stage-5.sch:20]: Indeed, the schematron is not generated correctly: the context is the parent XPath, but the "assert " are grouped by field. They should be grouped by parent XPath.
I'm working on correcting this.

Comment (bertrand-lorentz) [reply schematrons/stage-5.sch:19]: This has now been corrected with the commits added to this PR.

**Review by YvesJo — APPROVED** [review]
Looks good

**Pull request merged**